File: typescript/tests/stringCharacterFrequency.test.ts
import { describe, expect, it } from "vitest";

import { stringCharacterFrequency } from "../src/strings/stringCharacterFrequency.js";

describe("stringCharacterFrequency", () => {
  it("simple", () => {
    expect(stringCharacterFrequency("aab")).toEqual({ a: 2, b: 1 });
  });

  it("empty", () => {
    expect(stringCharacterFrequency("")).toEqual({});
  });

  it("all same", () => {
    expect(stringCharacterFrequency("aaa")).toEqual({ a: 3 });
  });

  it("space", () => {
    expect(stringCharacterFrequency("a a")).toEqual({ a: 2, " ": 1 });
  });
});

File: java/src/test/java/com/example/livecoding/strings/StringCharacterFrequencyTest.java
package com.example.livecoding.strings;

import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class StringCharacterFrequencyTest {

    @Test
    void simple() {
        Map<Character, Integer> m = StringCharacterFrequency.stringCharacterFrequency("aab");
        assertEquals(2, m.get('a'));
        assertEquals(1, m.get('b'));
        assertEquals(2, m.size());
    }

    @Test
    void empty() {
        assertTrue(StringCharacterFrequency.stringCharacterFrequency("").isEmpty());
    }

    @Test
    void allSame() {
        Map<Character, Integer> m = StringCharacterFrequency.stringCharacterFrequency("aaa");
        assertEquals(3, m.get('a'));
    }
}
